capa/features/extractors/dnfile/helpers.py

```diff
@@ -105,24 +107,13 @@ def get_dotnet_managed_imports(pe: dnfile.dnPE) -> Iterator[Tuple[int, str]]:
             TypeName (index into String heap)
             TypeNamespace (index into String heap)
     """
-    if not is_dotnet_table_valid(pe, "MemberRef"):
-        return
-
-    for (rid, row) in enumerate(pe.net.mdtables.MemberRef):
-        if not isinstance(row.Class.row, (dnfile.mdtable.TypeRefRow,)):
+    for (rid, row) in enumerate(iter_dotnet_table(pe, "MemberRef")):
+        if not isinstance(row.Class.row, dnfile.mdtable.TypeRefRow):
             continue
 
-        # like File::OpenRead
-        name = f"{row.Class.row.TypeName}::{row.Name}"
-
-        # ECMA II.22.38: TypeNamespace can be null or non-null
-        if row.Class.row.TypeNamespace:
-            # like System.IO.File::OpenRead
-            name = f"{row.Class.row.TypeNamespace}.{name}"
-
         token: int = calculate_dotnet_token_value(pe.net.mdtables.MemberRef.number, rid + 1)
 
-        yield token, name
+        yield token, row.Class.row.TypeNamespace, row.Class.row.TypeName, row.Name
```

Comment:
maybe use a class to group namespace, class, and method instead of passing around tuples?

Reply:
thank you for the suggestion! I've added these changes and the code is much cleaner. Please let me know what you think.